### ISSUE TITLE
Avoid_use - READ TABLE ... TRANSPORTING NO FIELDS

### ISSUE THREAD

**Issue by FreHu** [abaplint/abaplint#1319]
READ TABLE + subrc check can be replaced with `if line_exists( {{table_expression}}.`

**Comment by ed-holland** [comment]
Note that this function only exists starting 740.

**Comment by larshp** [comment]
https://github.com/SAP/styleguides/blob/master/clean-abap/CleanABAP.md#prefer-line_exists-to-read-table-or-loop-at

**Comment by larshp** [comment]
I'm starting this now

**Comment by larshp** [comment]
https://github.com/abaplint/abaplint/pull/1321

**Comment by larshp** [comment]
try again, 2.45.0 published, rule name = `use_line_exists`

**Comment by FreHu** [comment]
It's not triggered if the table variable has an unknown type.

**Comment by larshp** [comment]
hmm, the unit tests run without even the variables being defined

**Comment by FreHu** [comment]
Still not sure about the reason, but I have 46 occurrences of transporting no fields and only 14 results.

This one can be reproduced in playground:

```
DATA: itab TYPE STANDARD TABLE OF string.
READ TABLE itab TRANSPORTING NO FIELDS INDEX 1.
CHECK sy-subrc = 0.
```

**Comment by larshp** [comment]
also this one is not reported,

```abap
          READ TABLE rt_list TRANSPORTING NO FIELDS WITH KEY kind = 'E'.
          mv_success = boolc( sy-subrc <> 0 ).
```

the code only checks for IF, https://github.com/abaplint/abaplint/blob/master/packages/core/src/rules/use_line_exists.ts#L65

**Comment by FreHu** [comment]
Found the gotcha, if you have the subrc check within parentheses the rule doesn't work.

```
    IF ( sy-subrc = 0 ).
```

**Comment by larshp** [comment]
try again, 2.46.1 published
